Sources/NIOCore/EventLoop.swift

```diff
@@ -523,6 +523,33 @@ extension TimeAmount: AdditiveArithmetic {
     }
 }
 
+#if swift(>=5.7)
+extension TimeAmount {
+    @available(macOS 13, iOS 16, tvOS 16, watchOS 9, *)
+    /// Creates a new `TimeAmount` for the given ``Swift.Duration``.
+    ///
+    /// - parameters:
+    ///     - duration: ``Swift.Duration``.
+    /// - returns: the `TimeAmount` for the given amount, truncated to whole nanoseconds.
+    public static func from(_ duration: Swift.Duration) -> TimeAmount {
+        return .seconds(duration.components.seconds) + .nanoseconds(duration.components.attoseconds / (1_000_000_000))
+    }
+}
+
+@available(macOS 13, iOS 16, tvOS 16, watchOS 9, *)
+extension Swift.Duration {
+    /// Construct a `Duration` given a number of nanoseconds represented as a `TimeAmount`.
+    ///
+    ///       let t: TimeAmount = .seconds(1)
+    ///       let d: Duration = .from(t)
+    ///
+    /// - Returns: A `Duration` representing a given number of nanoseconds.
+    public static func from(_ timeAmount: TimeAmount) -> Duration {
```

Comment:
Same as above. I think this should be an init.

Reply:
Yes, I was back-and-forth on the spelling and wasn't happy enough to make this a non-draft.

I think I ended up with the static because I could use the existing `static Duration.nanoseconds(_:) -> Duration` as a shorthand but you're right ofc. I see that `Duration` has initialisers from `timeval` and `timespec` already. The best fit will probably be another one from `NIO.TimeAmount`. The many `1_000_000_000` were just making my eyes hurt :)